cirq/ion/ion_gates.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Operations native to iontrap systems."""
+
+import numpy as np
+
+from cirq import ops
+
+
+def MSGate(rads: float) -> ops.XXPowGate:
+    """
+    Implements the Mølmer–Sørensen gate,
+    native two-qubit operation in ion traps.
+    Effectively equivalents to rotation around the XX axis
+    in the two-qubit bloch sphere.
+
+    The gate implements the following unitary:
+    exp(-i t XX) = [cos(t) 0 0 -isin(t)]
+                   [0 cos(t) -isin(t) 0]
+                   [0 -isin(t) cos(t) 0]
+                   [-isin(t) 0 0 cos(t)]
+    rads = t, the rotation angle in radians
+    """
+    return ops.XXPowGate(exponent=rads*2/np.pi, global_shift=-0.5)
```

Comment:
It gets multiplied by 2? So MS(pi) isn't equal to Rx(pi) * Rx(pi)? Is that consistent with the matrix in the doc?

Reply:
MS(t) = exp(-i t XX) != exp(-i t/2 X) * exp(-i t/2 X).
I am using exponent=rads*2/np.pi because Pauli XX = -i exp(-i pi/2 XX)

Reply:
Okay, I see. That's a bit unusual, that rotating by pi only gets a you a quarter of the way through the global-phase-invariant cycle.

The number being passed in is still considered to be in radians, right? People won't be confused by the argument being called `rads` instead of, like `half_rads`?

We'll have to make sure that the diagram exponent and so forth are all consistent with this convention.

Reply:
Yes the number being passed in is still in radians.
RX(t) = exp(-i t/2 X) is the convention for single qubit gates because of the bloch sphere. I don't think such convention still stands for two-qubit gates. As long as people know they are doing MS(t) = exp(-i t XX) there should be no confusion.

Reply:
And I am fixing the diagram exponent for MS right now. No need to change the _diagram_exponent function.